source-gainsight-nxt/source_gainsight_nxt/resources.py

```diff
@@ -0,0 +1,196 @@
+        task: Task,
+        all_bindings,
+    ):
+        common.open_binding(
+            binding,
+            binding_index,
+            state,
+            task,
+            fetch_snapshot=functools.partial(
+                snapshot_resource,
+                http,
+                fields,
+                str(config.domain),
+                resource,
+                config.start_date,
+            ),
+            tombstone=GainsightResource(_meta=GainsightResource.Meta(op="d")),
+        )
+
+    resources = []
+    for name, resource in FULL_REFRESH_RESOURCES:
+        fields = await fetch_object_fields(http, log, str(config.domain), resource)
+
+        resources.append(
+            common.Resource(
+                name=name,
+                key=["/_meta/row_id"],
+                model=resource,
+                open=functools.partial(open, fields, resource),
+                initial_state=common.ResourceState(),
+                initial_config=common.ResourceConfig(
+                    name=name, interval=timedelta(minutes=5)
+                ),
+                schema_inference=True,
+            )
+        )
+
+    return resources
+
+
+async def incremental_resources(
+    log: Logger,
+    http: HTTPMixin,
+    config: EndpointConfig,
+) -> list[common.Resource]:
+    def open(
+        fields: list[str],
+        resource: type[GainsightResourceWithModifiedDate],
+        binding: CaptureBinding[common.ResourceConfig],
+        binding_index: int,
+        state: common.ResourceState,
+        task: Task,
+        all_bindings,
+    ):
+        common.open_binding(
+            binding,
+            binding_index,
+            state,
+            task,
+            fetch_page=functools.partial(
+                fetch_resource_page,
+                http,
+                fields,
+                str(config.domain),
+                resource,
+                config.start_date,
+            ),
+            fetch_changes=functools.partial(
+                fetch_resource_changes,
+                http,
+                fields,
+                str(config.domain),
+                resource,
+            ),
+        )
+
+    cutoff = datetime.now(tz=UTC)
+    resources = []
+    for name, resource in INCREMENTAL_RESOURCES:
+        fields = await fetch_object_fields(http, log, str(config.domain), resource)
+
+        resources.append(
+            common.Resource(
+                name=name,
+                key=["/Gsid"],
+                model=resource,
+                open=functools.partial(open, fields, resource),
+                initial_state=common.ResourceState(
+                    inc=common.ResourceState.Incremental(cursor=cutoff),
+                    backfill=common.ResourceState.Backfill(
+                        cutoff=cutoff, next_page=None
+                    ),
+                ),
+                initial_config=common.ResourceConfig(
+                    name=name, interval=timedelta(minutes=5)
```

Comment:
Does this need such a long interval? Usually for incremental bindings we would have this interval be much smaller (perhaps with no interval at all) unless there were rate limit considerations.

Reply:
Updated to 1 minute instead per our conversation via Slack.